src/main/java/org/folio/circulation/resources/RequestFromRepresentationService.java

```diff
@@ -150,51 +162,63 @@
   }
 
   private CompletableFuture<Result<Request>> fetchItemAndLoanForPageTlrRequest(Request request) {
-    if (request.getOperation() == Request.Operation.CREATE) {
-      return fromFutureResult(fetchItemForPageTlr(request)
-        .thenApply(r -> r.mapFailure(err -> errorHandler.handleValidationError(err,
-          NO_AVAILABLE_ITEMS_FOR_TLR, r))))
-        .flatMapFuture(this::fetchFirstLoanForUserWithTheSameInstanceId)
-        .flatMapFuture(this::fetchUserForLoan)
-        .toCompletableFuture();
-    } else {
-      return fromFutureResult(fetchItemForRequest(request))
-        .flatMapFuture(req -> succeeded(req).after(loanRepository::findOpenLoanForRequest)
-          .thenApply(r -> r.map(req::withLoan)))
-        .flatMapFuture(this::fetchUserForLoan)
-        .toCompletableFuture();
-    }
+    return request.getOperation() == Request.Operation.CREATE
+      ? fetchDataForPageTlr(request)
+      : fetchDataForNonPageTlr(request);
+  }
+
+  private CompletableFuture<Result<Request>> fetchDataForPageTlr(Request request) {
+    return fromFutureResult(completedFuture(fetchItemForPageTlr(request))
+      .thenApply(r -> r.mapFailure(err -> errorHandler.handleValidationError(err,
+        NO_AVAILABLE_ITEMS_FOR_TLR, r))))
+      .flatMapFuture(this::fetchFirstLoanForUserWithTheSameInstanceId)
+      .flatMapFuture(this::fetchUserForLoan)
+      .toCompletableFuture();
+  }
+
+  private CompletableFuture<Result<Request>> fetchDataForNonPageTlr(Request request) {
+    return fromFutureResult(fetchItemForRequest(request))
+      .flatMapFuture(req -> succeeded(req).after(loanRepository::findOpenLoanForRequest)
+        .thenApply(r -> r.map(req::withLoan)))
+      .flatMapFuture(this::fetchUserForLoan)
+      .toCompletableFuture();
+  }
+
+  private Result<Request> fetchItemForPageTlr(Request request) {
+    return getFirstAvailableItem(request)
+      .map(request::withItem)
+      .map(Result::succeeded)
+      .orElseGet(() -> failedValidation(
+        "Cannot create page TLR for this instance ID - no available items found", INSTANCE_ID,
+        request.getInstanceId()));
   }
 
-  private CompletableFuture<Result<Request>> fetchItemForPageTlr(Request request) {
-    return itemRepository.getFirstAvailableItemByInstanceId(request.getInstanceId())
-      .thenApply(r -> r.next(item -> {
-        if (item == null) {
-          return failedValidation(
-            "Cannot create page TLR for this instance ID - no available items found", INSTANCE_ID,
-            request.getInstanceId());
-        } else {
-          return succeeded(request.withItem(item));
-        }
-      }));
+  private Optional<Item> getFirstAvailableItem(Request request) {
```

Comment:
Do we actually use this method to get the first available item, or do we only need it to check if there is one? In other words, can it be boolean?